docs: Integration docs for Langchain4J and LangchainGo (#571)
qdrant/landing_page: qdrant-landing/content/documentation/frameworks/langchain4j.md

---
title: Langchain4J
weight: 2100
---

# Langchain4J

> [Langchain4J](https://github.com/langchain4j/langchain4j) is a community port of [Langchain](https://www.langchain.com/) for building context-aware AI applications in Java

Author: Langchain4J is a community port. Not affiliated with https://www.langchain.com/.

You can use Qdrant as a vectorstore in Langchain4J through the [`langchain4j-qdrant`](https://central.sonatype.com/artifact/dev.langchain4j/langchain4j-qdrant) module.

## Setup

> Add the `langchain4j-qdrant` to your project dependencies.

Author: So Java projects have multiple popular build tools to manage dependencies. This one is for the `pom.xml` file, if you're using Maven.

https://central.sonatype.com/artifact/dev.langchain4j/langchain4j-qdrant

<img width="950" alt="Screenshot 2024-02-01 at 21 04 43" src="https://github.com/qdrant/landing_page/assets/46051506/c840dc9b-ff35-4dc5-b5b2-ed55af6c1ff1">

> Add the `langchain4j-qdrant` to your project dependencies.

Reviewer: When reviewing this file, I didn't think "Oh Java!" So I'm suggesting a slight change to the title / intro

```xml
<dependency>
    <groupId>dev.langchain4j</groupId>
    <artifactId>langchain4j-qdrant</artifactId>
    <version>VERSION</version>
</dependency>
```

## Usage

> 

Reviewer: Similar to the other file:

I think you should highlight the variable code, to make sure readers know to substitute their configuration first. Here's how I might do this:

```suggestion

Before you use the following code sample, customize the following values for your configuration:

- `YOUR_COLLECTION_NAME`: Use our [Collections](/documentation/concepts/collections) guide to create or
  list collections.
- `YOUR_HOST_URL`: Use the REST URL for your system. If you used the [Quick Start](/documentation/quick-start/) guide,
  it may be http://localhost:6333. If you've deployed in the [Qdrant Cloud](/documentation/cloud/), you may have a
  longer URL such as `https://example.location.cloud.qdrant.io`.
- `YOUR_API_KEY`: Substitute the API key associated with your configuration.

Author: Since Langchain4J uses the Qdrant Java client through the GRPC interface.
https://github.com/qdrant/landing_page/pull/571/commits/898d18922eaa1f2770e47cc3a64ac54cf6206129

```java
import dev.langchain4j.store.embedding.EmbeddingStore;
import dev.langchain4j.store.embedding.qdrant.QdrantEmbeddingStore;

EmbeddingStore<TextSegment> embeddingStore =
    QdrantEmbeddingStore.builder()
        // Ensure the collection is configured with the appropriate dimensions
        // of the embedding model.
        // Reference https://qdrant.tech/documentation/concepts/collections/

>         .collectionName("{collection_name}")
>         .host("xyz-example.eu-central.aws.cloud.qdrant.io")
>         // GRPC port of the Qdrant server
>         .port(6334)
>         .apiKey("<YOUR_API_KEY")

Reviewer: I think we should have some consistency with the variables that we want readers to substitute for. And is `.host` limited to cloud hosts, or would it work with `http://localhost:6333`, based on the Quick Start guide (aka "YOUR_QDRANT_REST_URL," from the other file?)

So maybe:

(If you want to set it up in a different way, I'm good with that too. Long-term, I'd like to set up Hugo to substitute for those variables.)

```suggestion
        .collectionName("YOUR_COLLECTION_NAME")
        .host("YOUR_HOST_URL")
        // GRPC port of the Qdrant server
        .port(6334)
        .apiKey("YOUR_API_KEY")
```

---
title: Langchain4J
weight: 110
---

> # Langchain4J
> 
> [Langchain4J](https://github.com/langchain4j/langchain4j) is a community port of [Langchain](https://www.langchain.com/) for building context-aware AI applications in Java

Reviewer: Based on our discussion of maven... I didn't remember, "4J" refers to Java! So I'm thinking:

```suggestion
# LangChain for Java

LangChain for Java, also known as [Langchain4J](https://github.com/langchain4j/langchain4j), is a community port of [Langchain](https://www.langchain.com/) for building context-aware AI applications in Java
```
